services/director-v2/tests/conftest.py

```diff
@@ -61,10 +69,37 @@ def project_env_devel_dict(project_slug_dir: Path) -> Dict[str, Any]:
 def project_env_devel_environment(project_env_devel_dict: Dict[str, Any], monkeypatch):
     for key, value in project_env_devel_dict.items():
         monkeypatch.setenv(key, value)
+    monkeypatch.setenv(
+        "DYNAMIC_SIDECAR_IMAGE", "local/dynamic-sidecar:TEST_MOCKED_TAG_NOT_PRESENT"
+    )
+
+
+@pytest.fixture(scope="function")
+def mock_env(monkeypatch) -> None:
+    # Works as below line in docker.compose.yml
+    # ${DOCKER_REGISTRY:-itisfoundation}/dynamic-sidecar:${DOCKER_IMAGE_TAG:-latest}
+
+    registry = os.environ.get("DOCKER_REGISTRY", "local")
+    image_tag = os.environ.get("DOCKER_IMAGE_TAG", "production")
+
+    image_name = f"{registry}/dynamic-sidecar:{image_tag}".strip("/")
+    logger.warning("Patching to: DYNAMIC_SIDECAR_IMAGE=%s", image_name)
+    monkeypatch.setenv("DYNAMIC_SIDECAR_IMAGE", image_name)
+
+    monkeypatch.setenv("REGISTRY_auth", "false")
+    monkeypatch.setenv("REGISTRY_user", "test")
+    monkeypatch.setenv("REGISTRY_PW", "test")
+    monkeypatch.setenv("REGISTRY_ssl", "false")
+    monkeypatch.setenv("SIMCORE_SERVICES_NETWORK_NAME", "test_network_name")
+    monkeypatch.setenv("TRAEFIK_SIMCORE_ZONE", "test_traefik_zone")
+    monkeypatch.setenv("SWARM_STACK_NAME", "test_swarm_name")
+
+    monkeypatch.setenv("DIRECTOR_V2_DYNAMIC_SERVICES_ENABLED", "false")
+    monkeypatch.setenv("DIRECTOR_V2_MONITORING_ENABLED", "false")
```

Comment:
so I have to agree that monitoring is not very clearly defined in this context. it should at least be DYNAMIC_SERVICES_MONITORING... or SCHEDULER...

Reply:
ended up having longer names now

Reply:
by the way you can remove the REGISTRY entries I think. they are outdated now.

Reply:
done